### ISSUE TITLE
Support cases where .env doesn't exist.

### ISSUE THREAD

**Issue by christopher-francisco** [abacaphiliac/zend-phpdotenv#8]
I'm mostly working on setups where the `.env` file works as a quick environment variable definition helper for development purposes, but on QA/staging/production servers we mostly use the environment variables defined on the system itself.

It'd be great to support cases where if the .env doesn't exist, the system doesn't crash, kind of the way Laravel implements it on its [bootstrap process](https://github.com/laravel/framework/blob/5.3/src/Illuminate/Foundation/Bootstrap/DetectEnvironment.php#L24).

I'm willing to push a PR if this is OK, let me know 😄

**Comment by chris-fa** [comment]
However I have no idea how to test it; cant find a way to call onBootstrap on ModuleTest

**Comment by abacaphiliac** [comment]
yes this sounds like a great new feature. i can add some diagnostic checks in my applications that require the env file. i like the idea.

let me know if you have any questions or ideas about the unit tests : )

the existing ModuleTest triggers the ModuleManager, giving coverage over init and bootstrap. it's a collaboration test and pretty high level. up to you if you want to do a low and/or high level test.

**Comment by christopher-francisco** [comment]
Given `DotEnvFactory` re-throws the exception throw by `DotEnv` when there is no file found (`InvalidPathException`), I can't think of a different way other than removing those throws.

I tried using a boolean flag to support both cases, required and non-required .env; but the problem is that flag can't be injected from a config file (because dotenv should be initialized before the config files are merged, so that it can be read).

I'm kind of in a dead-end.

**Edit 1:** maybe said flag can be read from `application.config.php` ?

**Comment by abacaphiliac** [comment]
i think we're on the right track in our conversation on your request: #9 

here's how i'm thinking about solving the `.env` requirement/suggestion using Zend's diagnostic tools: #11
